test/gpu/octree/test_radius_search.cpp

```diff
@@ -57,6 +57,64 @@
 
 using namespace pcl::gpu;
 
+void
+verifyResults(std::vector<std::vector<int>>& host_search,
+                std::vector<std::vector<float>>& host_sqr_distances,
+                const NeighborIndices& device_results,
+                const Octree::ResultSqrDists& device_sqr_distances,
+                const std::vector<float>& radiuses,
+                const int max_answers,
+                const std::size_t query_size,
+                const int step)
+{
+    //download results
+    std::vector<int> sizes;
+    device_results.sizes.download(sizes);
+
+    std::vector<int> downloaded_buffer;
+    device_results.data.download(downloaded_buffer);
+
+    std::vector<float> downloaded_sqr_distances;
+    device_sqr_distances.download(downloaded_sqr_distances);
+
+    //verification
+    for(std::size_t i = 0; i < query_size; i+=step)
+    {
+        std::vector<int>& results_host = host_search[i];
+        std::vector<float>& sqr_dist_host = host_sqr_distances[i];
+
+        const int beg = i/step * max_answers;
+        const int end = beg + sizes[i/step];
+
+        const auto beg_dist2 = downloaded_sqr_distances.cbegin() + beg;
+        const auto end_dist2 = downloaded_sqr_distances.cbegin() + end;
+
+        std::vector<int> results_batch (downloaded_buffer.cbegin() + beg, downloaded_buffer.cbegin() + end);
+        const std::vector<float> sqr_distances_batch (beg_dist2, end_dist2);
+
+        std::sort(results_batch.begin(), results_batch.end());
+        std::sort(results_host.begin(), results_host.end());
+
+        if (results_batch.size() == max_answers && results_batch.size() < results_host.size() && max_answers)
+        {
+            results_host.resize(max_answers);
+            sqr_dist_host.resize(max_answers);
+        }
+        const float sqr_radius = radiuses[i] * radiuses[i];
+        for (std::size_t j = 0; j < sqr_dist_host.size(); j++)
+        {
+            ASSERT_LT (sqr_distances_batch[j], sqr_radius);
+            ASSERT_NEAR (sqr_distances_batch[j], sqr_dist_host[j], 0.001);
+        }
+        ASSERT_EQ ((results_batch == results_host), true);
+    }
+
+    const float avg_size = std::accumulate(sizes.begin(), sizes.end(), 0) * (1.f/sizes.size());
+    std::cout << "avg_result_size = " << avg_size << "\n";
+
+    ASSERT_GT(avg_size, 5);
```

Comment:
What's the purpose of this test? I doesn't look deterministic at all.

Reply:
Yes it doesn't seem to prove anything concrete. It seems to be checking that the average number of points identified are over 5. This is naturally going to be dependent on the params used to initialize the input cloud.

Reply:
Shall I get rid of it?

Reply:
yes please